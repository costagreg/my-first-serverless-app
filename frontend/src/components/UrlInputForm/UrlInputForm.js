import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import useInput from '../../hooks/useInput'

import './UrlInputForm.scss'

const UrlInputForm = (props) => {
  const { value, bindInput } = useInput('')
  const { createUrl, error } = props

  const handleSubmit = (evt) => {
    createUrl(value)
    evt.preventDefault()
  }

  return (
    <form className="urlInputForm" onSubmit={handleSubmit}>
      <input
        className={classNames('urlInputForm__input', {
          'urlInputForm__input--error': error,
        })}
        type="text"
        placeholder="https://www.example.com"
        {...bindInput}
      />
      <input className="urlInputForm__submit" value="Cut me!" type="submit" />
    </form>
  )
}

UrlInputForm.propTypes = {
  createUrl: PropTypes.func.isRequired,
  error: PropTypes.string,
}

export default UrlInputForm
