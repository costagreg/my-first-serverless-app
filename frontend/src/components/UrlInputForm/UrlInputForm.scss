.urlInputForm{
  display: flex;
  justify-content: center;
  align-items: center;
  width: 50%;
  margin: 40px auto;
  

  &__input{
    flex: 1;
    border: 1px solid #cecece;
    border-radius: 5px;
    height: 40px;
    margin: 0 10px 0 0;

  }

  &__submit{
    box-sizing: content-box;
    border: 1px solid #cecece;
    border-radius: 5px;
    height: 40px;
  }
}