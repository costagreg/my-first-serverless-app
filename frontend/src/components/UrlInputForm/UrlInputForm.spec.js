import React from 'react'
import UrlInputForm from './UrlInputForm'
import { render, fireEvent } from '@testing-library/react'

describe('UrlInputForm', () => {
  describe('@renders', () => {
    it('renders UrlInputForm without any error', () => {
      const { asFragment } = render(<UrlInputForm />)

      expect(asFragment()).toMatchSnapshot()
    })

    describe('error is passed as prop', () => {
      it('adds a classname to input', () => {
        const { getByRole } = render(<UrlInputForm error='Url not valid' />)

        const urlInput = getByRole('textbox')

        expect(urlInput).toHaveClass('urlInputForm__input--error')
      })
    })
  })

  describe('@submit', () => {
    it('should call createUrl prop', () => {
      const createUrl = jest.fn(() => {})
      const { getByRole } = render(
        <UrlInputForm createUrl={createUrl} />
      )

      const urlInput = getByRole('textbox')
      const submitButton = getByRole('button')

      fireEvent.change(urlInput, {
        target: { value: 'jshaksj' },
      })

      fireEvent.click(submitButton)

      expect(createUrl).toHaveBeenCalledTimes(1)
    })
  })
})
