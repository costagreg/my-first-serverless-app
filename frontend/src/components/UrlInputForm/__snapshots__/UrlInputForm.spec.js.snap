// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlInputForm @renders renders UrlInputForm without any error 1`] = `
<DocumentFragment>
  <form
    class="urlInputForm"
  >
    <input
      class="urlInputForm__input"
      placeholder="https://www.example.com"
      type="text"
      value=""
    />
    <input
      class="urlInputForm__submit"
      type="submit"
      value="Cut me!"
    />
  </form>
</DocumentFragment>
`;

exports[`UrlInputForm @renders renders UrlInputFormwithout any error 1`] = `
<DocumentFragment>
  <form
    class="urlInputForm"
  >
    <input
      class="urlInputForm__input"
      placeholder="https://www.example.com"
      type="text"
      value=""
    />
    <input
      class="urlInputForm__submit"
      type="submit"
      value="Cut me!"
    />
  </form>
</DocumentFragment>
`;
